Lesson27-Loops&Conditions/scripts/main.js

```diff
@@ -0,0 +1,301 @@
+            }    
+        //  Check for negative numbers
+            if (userInputArray[i] < 0) {
+                negativeNumbers.push(userInputArray[i])            
+            }     
+        //  Check for zero numbers
+            if (userInputArray[i] === 0) {
+                zeroNumbers.push(userInputArray[i])            
+            }  
+        // // Check for even numbers       
+            if (userInputArray[i] % 2 === 0) {
+                evenNumbers.push(userInputArray[i])            
+            }
+        // // Check for odd numbers       
+        if (userInputArray[i] % 2 != 0) {
+            oddNumbers.push(userInputArray[i])            
+        }    
+    }
+    // Printing statements 
+        console.log('Your entered list was ' + userInputArray);
+        console.log('All the positive numbers in your list is: ' + positiveNumbers);
+        console.log('The length of your positive number list is: ' + positiveNumbers.length);
+        console.log('All the negative numbers in your list is: ' + negativeNumbers);
+        console.log('The length of your negative number list is: ' + negativeNumbers.length);
+        console.log('All the zero numbers in your list is: ' + zeroNumbers);
+        console.log('The length of your zero number list is: ' + zeroNumbers.length);
+        console.log('All the even numbers in your list is: ' + evenNumbers);
+        console.log('The length of your even number list is: ' + evenNumbers.length);
+        console.log('All the odd numbers in your list is: ' + oddNumbers);
+        console.log('The length of your odd number list is: ' + oddNumbers.length);
+    }
+    // numberStats()
+
+
+// 6. Loop a calculator. Request 2 numbers and a sign, solve the problem, display the result and ask if the user wants one more.
+//      The loop continues until the user refuses. 
+//[[BUGG: MULTIPLY DOESN'T MULTIPLY]]
+function loopCalculator() {
+    do {
+        let numberX = parseInt(prompt('Please enter a number x:'))
+        let numberY = parseInt(prompt('Please enter a number y:'))
+        let sign = prompt('Please choose a sign: +, -, /, * ')
+    
+        if (sign == '+') {
+            let addSum = numberX + numberY;
+            console.log('The calculation of x and y is: ' + addSum);
+            question = prompt('Would you like to play again?');
+            console.log(question);
+            continue;            
+        }
+        if (sign == '-'); {
+            let minusSum = numberX - numberY;
+            console.log('The calculation of x and y is: ' + minusSum);
+            question = prompt('Would you like to play again?');
+            console.log(question);
+            continue;
+        }
+        if (sign == '/'); {
+            let devideSum = numberX / numberY;
+            console.log('The calculation of x and y is: ' + devideSum);
+            question = prompt('Would you like to play again?');
+            continue;
+        }
+        if (sign == '*'); {
```

Comment:
Previously you used '==' but here you using '==='
What is the difference?

Reply:
== > convert the values it compares to the same type and then compare it and 
=== > is a strict operator. It won't convert the values before comparing them, it will return false if the values are not the same type.
I'm just not sure which one is the more general/common one to use?